### PR TITLE
Fix overflow issue in sequence

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue: https://github.com/microsoft/Power-Fx/issues/1368

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs [hunk 1]
```diff
@@ -827,33 +827,75 @@ public static FormulaValue Sequence(IRContext irContext, FormulaValue[] args)
...
+                {
+                    yield return new ErrorValue(irContext, new ExpressionError()
+                    {
+                        Message = x > 0 ? $"Number is larger than maximum double." : $"Number is less than minimum double.",
+                        Span = irContext.SourceContext,
+                        Kind = ErrorKind.Numeric
+                    });
+                }
+
+                if (isValid && ((double)(x / 2) + (double)(step / 2) <= (double)(double.MaxValue / 2)) && ((double)(x / 2) + (double)(step / 2) >= (double)(double.MinValue / 2)))
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:860]: >((double)(x / 2) + (double)(step / 2) <= (double)(double.MaxValue / 2)) && ((double)(x / 2) + (double)(step / 2) >= (double)(double.MinValue / 2))) [](http://example.com/codeflow?start=31&length=147)

This check can be simplified by letting C# check the boundaries for you:

    if (isValid) {
        x += step;
        isValid = double.IsFinite(x);
    }

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs [hunk 2]
```diff
@@ -827,33 +827,75 @@ public static FormulaValue Sequence(IRContext irContext, FormulaValue[] args)
...
+                {
+                    yield return new ErrorValue(irContext, new ExpressionError()
+                    {
+                        Message = x > 0 ? $"Number is larger than maximum decimal." : $"Number is less than minimum decimal.",
+                        Span = irContext.SourceContext,
+                        Kind = ErrorKind.Numeric
+                    });
+                }
+
+                if (isValid && ((decimal)(x / 2) + (decimal)(step / 2) <= (decimal)(decimal.MaxValue / 2)) && ((decimal)(x / 2) + (decimal)(step / 2) >= (decimal)(decimal.MinValue / 2)))
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:891]: >if (isValid && ((decimal)(x / 2) + (decimal)(step / 2) <= (decimal)(decimal.MaxValue / 2)) && ((decimal)(x / 2) + (decimal)(step / 2) >= (decimal)(decimal.MinValue / 2))) [](http://example.com/codeflow?start=16&length=170)

This check can be simplified by letting C# check the boundaries for you:

    if (isValid) {
        try {
            x += step;
        } catch(OverflowException) {
            isValid = false;
        }
    }

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs [hunk 3]
```diff
@@ -827,33 +827,75 @@ public static FormulaValue Sequence(IRContext irContext, FormulaValue[] args)
...
-                x += step;
+                if (isValid)
+                {
+                    yield return new NumberValue(IRContext.NotInSource(FormulaType.Number), x);
+                }
+                else
+                {
+                    yield return new ErrorValue(irContext, new ExpressionError()
+                    {
+                        Message = x > 0 ? $"Number is larger than maximum double." : $"Number is less than minimum double.",
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:854]: >Number is larger than maximum double." : $"Number is less than minimum double. [](http://example.com/codeflow?start=44&length=78)

Suggestion: "The expression returned a non-finite number." - same error message as JS produces when an operation generates an overflow

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs [hunk 4]
```diff
@@ -827,33 +827,75 @@ public static FormulaValue Sequence(IRContext irContext, FormulaValue[] args)
...
+            {                
+                if (isValid)
+                {
+                    yield return new DecimalValue(IRContext.NotInSource(FormulaType.Decimal), x);
+                }
+                else
+                {
+                    yield return new ErrorValue(irContext, new ExpressionError()
+                    {
+                        Message = x > 0 ? $"Number is larger than maximum decimal." : $"Number is less than minimum decimal.",
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:885]: >"Number is larger than maximum decimal." : $"Number is less than minimum decimal." [](http://example.com/codeflow?start=43&length=82)

Suggestion: "The expression returned a number that is beyond the range of decimal values"

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs (hunk 1 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs]: updated. thanks

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs (hunk 2 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs]: updated

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs (hunk 3 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs]: updated.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs (hunk 4 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs]: updated

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

**Pull request merged**